recipes/opentelemetry-cpp/all/conanfile.py

```diff
@@ -53,19 +86,57 @@ def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("abseil/20220623.0")
-        self.requires("grpc/1.50.1")
-        self.requires("libcurl/7.86.0")
-        self.requires("nlohmann_json/3.11.2")
-        self.requires("openssl/1.1.1s")
-        if Version(self.version) <= "1.4.1":
-            self.requires("opentelemetry-proto/0.11.0")
-        else:
-            self.requires("opentelemetry-proto/0.19.0")
-        self.requires("protobuf/3.21.4")
-        self.requires("thrift/0.17.0")
-        if Version(self.version) >= "1.3.0":
-            self.requires("boost/1.80.0")
+        requires = set()
+
+        if self.options.with_gsl:
+            requires.add("ms-gsl/4.0.0")
+
+        if self.options.with_abseil:
+            requires.add("abseil/20220623.0")
+
+        if self.options.with_otlp:
+            requires.add("protobuf/3.21.4")
+            if Version(self.version) <= "1.4.1":
+                requires.add("opentelemetry-proto/0.11.0")
+            else:
+                requires.add("opentelemetry-proto/0.19.0")
+
+            if self.options.with_otlp_grpc:
+                requires.add("grpc/1.50.1")
+
+            if self.options.with_otlp_http:
+                requires.add("libcurl/7.87.0")
+                requires.add("nlohmann_json/3.11.2")
+                requires.add("openssl/1.1.1s")
+
+        if self.options.with_zipkin:
+            requires.add("nlohmann_json/3.11.2")
+            requires.add("libcurl/7.87.0")
+            requires.add("openssl/1.1.1s")
+
+        if self.options.with_prometheus:
+            requires.add("prometheus-cpp/1.1.0")
+
+        if self.options.with_elasticsearch:
+            requires.add("nlohmann_json/3.11.2")
+            requires.add("libcurl/7.87.0")
+            requires.add("openssl/1.1.1s")
+
+        if self.options.with_jaeger:
+            requires.add("thrift/0.17.0")
+
+            if Version(self.version) >= "1.3.0":
+                if self.settings.compiler == "apple-clang":
+                    requires.add("boost/1.80.0")
+                else:
+                    requires.add("boost/1.81.0")
+
+        if self.options.with_etw:
+            requires.add("nlohmann_json/3.11.2")
+            requires.add("openssl/1.1.1s")
+
+        for r in requires:
+            self.requires(r)
 
     def validate(self):
```

Comment:
There should be a check here that boost::local is enabled

Reply:
I'm sorry, I don't quite understand this suggestion 😅

Reply:
Sorry reviews when I am on my mobile are a little more self serve 🙊 here's what I was thinking... it a kinda common thing for boost dependencies

https://github.com/conan-io/conan-center-index/blob/b2beeacb7353aa3fc20a4e6d04039dab682559bf/recipes/gtsam/all/conanfile.py#L95-L105

Reply:
This will block if the dep is not available in the current graph

Reply:
Thanks, this approach is working.